File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
""" define a class for rectangle """


class Rectangle:

    def __init__(self, width=0, height=0):
        self.height = height
        self.width = width

    @property
    def width(self):
        """getter for private instance width"""
        return(self.__width)

    @width.setter
    def width(self, value):
        """setter for private instance width"""
        if type(value) is not int:
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        """getter for private instance height"""
        return (self.__height)

    @height.setter
    def height(self, value):
        """setter for private instance height"""
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        return self.__width * self.__height

    def perimeter(self):
        return 2*(self.__width) + 2*(self.__height)

    def __str__(self):
        """returns printable string representation of the rectangle"""
        string = ""
        if self.__width != 0 and self.__height != 0:
            string += "\n".join("#" * self.__width
                                for j in range(self.__height))
            return string

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node

// class rectangle
class Rectangle {
  constructor (w, h) {
    if (h >= 0 || w >= 0) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    const char = 'X';
    for (let col = 0; col < this.height; col++) {
      console.log(char.repeat(this.width));
    }
  }
}
module.exports = Rectangle;
